app/wallet/vsp.js

```diff
@@ -50,9 +53,44 @@ export const getVSPTicketStatus = promisifyReqLogNoData(
   api.getTicketStatus
 );
 
-export const getAllVSPs = promisifyReqLogNoData(
-  "getAllVspsInfo",
-  api.getAllVspsInfo
+// addAllowedVSPsInCfg modifies the config file to allow the given VSP hosts
+// to be accessed. Returns true if the list of allowed hosts changed.
+const addAllowedVSPsInCfg = (hosts) => {
+  const c = cfg.getGlobalCfg();
+  const newHosts = c.get(cfgConstants.ALLOWED_VSP_HOSTS);
+  const oldHostsLen = newHosts.length;
+  hosts.forEach(vsp => newHosts.indexOf(vsp) === -1 && newHosts.push(vsp));
+  if (oldHostsLen !== newHosts.length) {
+    c.set(cfgConstants.ALLOWED_VSP_HOSTS, newHosts);
+    return true;
+  }
+
+  return false;
+};
+
+export const getAllVSPs = withLogNoData(
+  async () => {
+    const res = await new Promise((ok, fail) =>
+      api.getAllVspsInfo((res, err) => (err ? fail(err) : ok(res)))
+    );
+
+    // Allow access to all VSPs returned by the official VSP listing endpoint.
+    // This is less then ideal because this endpoint might eventually return
+    // domains that switched owners or that are otherwise available for hijack,
+    // but is needed due to Decrediton (currently) having to iterate over all
+    // existing VSPs to discover their pubkey and sync tickets.
+    //
+    // Eventually this can be further locked down once that iteration isn't
+    // performed any more and VSPs are only accessed when attempting to purchase
+    // a ticket.
```

Comment:
Surfacing issue. Ideally the public VSP listing API should be providing all the basic information we need such that we _don't_ need to be reaching out to every VSP on every wallet start to figure out this info.

Need to double check whether the VSP public key is the only thing missing or if there's anything else.